src/securejoin.rs

```diff
@@ -1001,25 +1001,25 @@ async fn mark_peer_as_verified(
 fn encrypted_and_signed(
     context: &Context,
     mimeparser: &MimeMessage,
-    expected_fingerprint: impl AsRef<str>,
+    expected_fingerprint: Option<&Fingerprint>,
```

Comment:
The old code wanted to accept the empty string, this is replaced with an option.